src/org/exist/indexing/IndexManager.java

```diff
@@ -111,11 +109,20 @@ private AbstractIndex initIndex(BrokerPool pool, String id, Element config, Path
     public Index registerIndex(Index index) throws DatabaseConfigurationException {
         index.open();
         indexers.put(index.getIndexId(), index);
-        if (LOG.isInfoEnabled())
-            {LOG.info("Registered index " + index.getClass() + " as " + index.getIndexId());}
+        if (LOG.isInfoEnabled()) {
+            LOG.info("Registered index " + index.getClass() + " as " + index.getIndexId());
+        }
         return index;
     }
 
+    public void unregisterIndex(Index index) throws DBException {
+        indexers.remove(index.getIndexId(), index);
```

Comment:
Should we call `Index#close` or should we expect the caller to have already called it?